controllers/admin/AdminCartRulesController.php

```diff
@@ -200,10 +200,44 @@ public function postProcess()
                 }
             }
 
-            // These are checkboxes (which aren't sent through POST when they are not check), so they are forced to 0
+            // These are checkboxes (which aren't sent through POST when they are not checked), so they are forced to 0
+            $restrictions_selected = false;
             foreach (['country', 'carrier', 'group', 'cart_rule', 'product', 'shop'] as $type) {
                 if (!Tools::getValue($type . '_restriction')) {
                     $_POST[$type . '_restriction'] = 0;
+                } elseif ($type !== 'shop') { // shop restriction is forced to be checked, thus we do not take it into account
+                    $restrictions_selected = true;
+                }
+            }
+
+            // At least one of these is compulsory: code, customer or any restriction. If none of these are present, the cart rule applies to all carts !
+            if (empty(Tools::getValue('id_cart_rule')) &&
+                empty($restrictions_selected) &&
+                empty(Tools::getValue('id_customer')) && empty(Tools::getValue('code')) && empty(Tools::getValue('minimum_amount'))
+                && Tools::getValue('date_from') < date('Y-m-d H:i:s')
```

Comment:
Is there a problem using `PrestaShop\PrestaShop\Core\Util\DateTime\DateTime::DEFAULT_DATETIME_FORMAT` ? Is it deprecated class ?

(sorry if questions sounds stupid but I am still a noob in Prestashop project and I am always suspicious with hard coded formatting ^^')

Reply:
How do you manage Timezone ?
I mean if `date_from` is given UTC+12 and server `date(...)` is UTC+2, don't we risk a "start time in the past" situation occurs ?

Is `date_from` given by server ? (I don't trust user/front data ^^')
Is it checked somewhere ?

Reply:
> How do you manage Timezone ?

I consider this out of scope. I don't know exactly how cart rules manage time zones. But what ever is the way it is managed or not, I don't think about changing it.

> Is date_from given by server ? (I don't trust user/front data ^^')

It results from a datepicker in the BO, so it must be a valid date.
The column in mysql is `datetime`, so it will not be possible to save something that isn't a date. As such it is validated on saving through https://github.com/PrestaShop/PrestaShop/blob/116a071edfa09c82df5d760ebd662be7fde8d09a/classes/CartRule.php#L118

Reply:
> > How do you manage Timezone ?
> 
> I consider this out of scope. I don't know exactly how cart rules manage time zones. But what ever is the way it is managed or not, I don't think about changing it.

:+1: It was more questions for my own knowledge.
I think this way to manage datetime could induce lot of troubles for advanced usage of Prestashop :thinking: I keep this comment in my mind and I'll investigate more ^^

> > Is date_from given by server ? (I don't trust user/front data ^^')
> 
> It results from a datepicker in the BO, so it must be a valid date. The column in mysql is `datetime`, so it will not be possible to save something that isn't a date. As such it is validated on saving through

BO is still a frontend so I don't trust it but if it is ok for me if it is validated before to be used or stored (which should avoid unpredictable error for end user and possible security issues).

Thanks for your answers ^^